io/src/main/scala/fs2/io/file/package.scala

```diff
@@ -0,0 +1,56 @@
+package fs2.io
+
+import java.nio.ByteBuffer
+import java.nio.channels.{AsynchronousFileChannel, CompletionHandler}
+import java.nio.file.{OpenOption, Path, StandardOpenOption}
+
+import fs2._
+import fs2.util.Monad
+
+package object file {
+
+  private[fs2] def asyncCompletionHandler[F[_], O](f: CompletionHandler[O, Null] => F[Unit])(implicit F: Async[F]): F[O] = {
+    F.async[O] { cb =>
+      f(new CompletionHandler[O, Null] {
+        override def completed(result: O, attachment: Null): Unit = cb(Right(result))
+
+        override def failed(exc: Throwable, attachment: Null): Unit = cb(Left(exc))
+      })
+    }
+  }
+
+  // uses default NIO thread pool for callbacks... okay? should be configurable?
+  def open[F[_]](path: Path, flags: Seq[OpenOption])(implicit F: Monad[F]): Pull[F, Nothing, AsynchronousFileChannel] = {
```

Comment:
Do you think we will want non-NIO based file primitives? If not, then maybe drop the `Async` suffix from the read/write methods. If so, then maybe add `Async` suffix here?